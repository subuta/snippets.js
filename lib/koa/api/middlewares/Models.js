import { build, format, snippets as s } from 'bld.js'

export default () => {
  const imports = s.import([
    ['src/api/model', 'models']
  ])

  return build`
    ${imports}
    
    ${s.export(build`
      (ctx, next) => {
        // expose Objection.js models to context.
        ctx.state.models = models
        return next()
      }
    `)}
  `
}
