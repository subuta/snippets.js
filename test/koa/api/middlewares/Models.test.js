import test from 'ava'
import { build, format, snippets as s } from 'bld.js'

import Models from 'lib/koa/api/middlewares/Models'

test.beforeEach(async (t) => {

})

test.afterEach((t) => {
})

test('should create koa Auth', async (t) => {
  const code = Models()

  const expected = build`
    import models from 'src/model'

    export default (ctx, next) => {
      // expose Objection.js models to context.
      ctx.state.models = models
      return next()
    }
  `

  t.is(format(code), format(expected))
})