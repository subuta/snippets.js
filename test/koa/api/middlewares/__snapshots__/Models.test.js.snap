// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`koa/api/middlewares/Models should create koa Models 1`] = `
"import models from 'src/api/model'

export default (ctx, next) => {
  // expose Objection.js models to context.
  ctx.state.models = models
  return next()
}
"
`;
